File: app/src/main/java/com/stocardapp/hackschoolchat/database/ChatDao.java
package com.stocardapp.hackschoolchat.database;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;


@Dao
public interface ChatDao {
    @Query("SELECT * FROM chatmessage")
    List<ChatMessage> getAll();

    @Query("SELECT * FROM chatmessage")
    LiveData<List<ChatMessage>> getAllLive();

    @Query("SELECT * FROM chatmessage WHERE uid IN (:userIds)")
    List<ChatMessage> loadAllByIds(int[] userIds);

    @Query("SELECT * FROM chatmessage WHERE name LIKE :name")
    ChatMessage findByName(String name);

    @Insert
    void insert(ChatMessage messages);

    @Insert
    void insertAll(List<ChatMessage> messages);

    @Delete
    void delete(ChatMessage user);

    @Query("DELETE FROM chatmessage")
    void nukeTable();
}

File: app/src/main/java/com/stocardapp/hackschoolchat/database/ChatDao.kt
package com.stocardapp.hackschoolchat.database

import androidx.lifecycle.LiveData
import androidx.room.*


@Dao
interface ChatDao {

    @Query("SELECT * FROM chats")
    fun getAllLive(): LiveData<List<ChatMessage>>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(messages: ChatMessage)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertAll(messages: List<ChatMessage>)

    @Delete
    fun delete(user: ChatMessage)

    @Query("DELETE FROM chats")
    fun nukeTable()
}
